source/documentation/monitoring-an-app/application-metrics.md

```diff
@@ -1,107 +1,121 @@
 
-Click [here](https://github.com/prometheus/docs/blob/master/content/docs/instrumenting/exporters.md) for a list of exporters and client libraries listed on the official Prometheus Github repo.
+#### Changing the application code
+The [application](https://github.com/ministryofjustice/cloud-platform-multi-container-demo-app) we're using in this example is a simple (by its own definition) rails application, which we'll have installed in our own namespace. As we're using Ruby we'll need to install the `prometheus-client `gem that will eventually give us a `/metric` endpoint. 
 
+First, simply add the gem to your Gemfile and install with bundler (if your following on with your own application, you may need to install the `rack` middleware gem). 
 
-#### Instrumentation of The Cloud-Platform Reference Application
-The [Cloud Platform Reference Application](https://github.com/ministryofjustice/cloud-platform-reference-app) has `instrumentation` setup using [django-prometheus](https://github.com/korfuri/django-prometheus). The default install steps were followed which created a metrics endpoint.
+```
+gem `prometheus-client`
+```
 
-See screenshot below of how the metrics endpoint looks on a browser:
-![Image of metrics](https://raw.githubusercontent.com/ministryofjustice/cloud-platform-user-docs/master/images/metrics-endpoint.png)
+Next we need to amend the `config.ru` file and inclued the two `rack` middlewares required by the `prometheus-client`. 
 
+```
+require_relative 'config/environment'
+require 'prometheus/middleware/collector'
+require 'prometheus/middleware/exporter'
 
-#### Create a Service to expose Pods
+use Prometheus::Middleware::Collector
+use Prometheus::Middleware::Exporter
 
-Scraping an exporter or separate metrics port requires a service that targets the Pod(s) of the exporter or application.
+run Rails.application
+```
 
-Example:
+If you're running this locally, you'll now be able to query your `/metrics` endpoint and display an output. 
 
-```yaml
-kind: Service
-apiVersion: v1
-metadata:
-  name: my-app-service
-  labels:
-    app: my-app
-spec:
-  selector:
-    app: my-app
-  ports:
-  - protocol: TCP
-    port: 8000
-    targetPort: 8000
-    name: metrics
-  type: NodePort
+```
+curl localhost:3000/metrics
 ```
 
-#### Create a Service-Monitor
-A `ServiceMonitor` is a resource the Prometheus Operator introduces for Kubernetes that describes the set of targets to be monitored by Prometheus
+Build, tag and push your application changes to your code repository and deploy the latest version into your Cloud Platform namespace. Confirm your `/metrics` endpoint is now accessible from your url. 
 
-Service Monitor Architecture
-![Image of Service-Monitor Architecture](https://raw.githubusercontent.com/ministryofjustice/cloud-platform-user-docs/master/images/service-monitor-arch.png)
+```
+curl https://myapp.cloud-platform/metrics
+```
 
```

Comment:
What should the output look like, if all is well?

Reply:
Added a description of output if fails.